libdvbtee/parse.cpp

```diff
@@ -1158,7 +1158,16 @@ bool parse::is_psip_ready()
 
 bool parse::is_epg_ready()
 {
-	return ((is_psip_ready()) && ((decoders.count(get_ts_id()) && (decoders[get_ts_id()].got_all_eit(eit_collection_limit)))));
+	if (!is_psip_ready()) return false;
+	if (!decoders.count(get_ts_id())) return false;
+	decode& d=decoders[get_ts_id()];
+	if (!d.got_all_eit(eit_collection_limit)) return false;
+#if ETT
+#if 1
+	if (!d.got_all_ett()) return false;
```

Comment:
This is going to need to be conditionalized.  I can imagine cases where the user may want to know a quick list of all shows currently airing without the need for collecting all the events long descriptions.

EIT's are conditionalized by the `eit_collection_limit`, which might be zero.  For ETT I think we just need a boolean, and your function seems to have the right idea comparing received ETTs against expected ETTs.